### ISSUE TITLE
Ap probably doesn't mark par as completed

### ISSUE THREAD

**Issue by mikevoronov** [fluencelabs/aquavm#178]
From a user request, the following AIR script 

```
(seq
    (par
        (call relay ("peer" "timeout") [1 "slow_result"] $result)
        (ap "fast_result" $result)
    )
    (call %init_peer_id% ("op" "return") [$result.$[0])
)
```

doesn't send data to `%init_peer_id%` after execution of `peer timeout`. This is probably because of a `ap` instruction doesn't mark a `par` block as completed.

**Comment by mikevoronov** [comment]
It's turned out that this isn't a bug, but an expected behaviour of the interpreter. Tree completeness in the interpreter is figured out by the following algo:
- par/seq set the `subtree_complete` boolean flag to true
- each instruction if it's incomplete (such as `call`/`match`/`mismath` while joining) set `subtree_complete` to false to represent this fact
- after execution of each subtree `seq`/`par` determines `subtree_complete` and then works in an appropriate way
- `ap` instructions doesn't touch this flag

So, in the following script the last call will be executed in the first interpeter run.

**Comment by mikevoronov** [comment]
Closed by #175.
